docs/scripting/language/Tags.md

```diff
@@ -0,0 +1,170 @@
+---
+id: Tags
+title: "Scripting: Statements"
+---
+
+## `Introduction`
+A tag is a prefix to a variable which tells the compiler to treat the variable specially under certain circumstances. For example you can use tags to define where a variable can and can't be used, or a special way to add two variables together.
+
+There are two types of tag - strong tags (starting with a capital letter) and weak tags (starting with a lower case letter), for the most part they're the same however under certain circumstances weak tags can be converted to tagless silently by the compiler, i.e. you won't get a warning, most of the time with weak tags, and all the time with strong tags, implicitly changing the tag will result in a warning to tell you data is likely being used wrong.
+
+A very simple example is the following:
+```c
+new
+    File:myfile = fopen("file.txt", io_read);
+myFile += 4;
+```
+
+The fopen function will return a value with a tag of type "File:", there is no problem on that line as the return value is being stored to a variable also with a tag of "File:" (note the cases are the same too). However on the next line the value 4 is added to the file handle. 4 has no tag (it is actually tag type "_:" but variables, values and functions with no tag are automatically set to that and you don't need to worry about it normally) and myFile has a tag of "File:", obviously nothing and something can't possibly be the same so the compiler will issue a warning, this is good as a handle to a file is meaningless in terms of it's actual value and so modifying it will merely destroy the handle and mean the file can't be closed as there is no longer a valid handle to pass and close the file with.
```

Comment:
This doesn't matter a _whole_ lot and I'm not going to block the merge over it, but a really nice readability QoL thing for technical documents would be to surround any reference to technical symbols with a `code` block. So `fopen` and `File:` and `4` and `_:` it makes it easier to associate these elements with larger code examples and when scanning large blocks of text, it allows you quickly understand which words are English and which words are technical elements.

Reply:
Done